### DISCUSSION TITLE
Macros

### DISCUSSION THREAD

**Discussion by remote110** [bassamanator/Sovol-SV06-firmware#68] [Q&A] [answered]
Hello, I'm trying to use this on the Biqu pad 7.
I run into an issue where if I enable the below
[include ./cfgs/misc-macros.cfg]
[include ./cfgs/PARKING.cfg]
[include ./cfgs/MECHANICAL_GANTRY_CALIBRATION.cfg]
It breaks my ability to open files. 
I can upload to the device fine but after the upload it gives an error that it's unable to open the file. 
Have you seen this before?

**Comment by bassamanator** [comment]
So, you downloaded and setup this config on the biqu pad yourself?

**Reply by remote110** [reply]
I did. 
And then I had this problem and just switched back to kiauh and used your sv06 plus printer.cfg info. 
I moved the cfgs folder to the proper location once I figured that part out and the macros showed up. 
But, like I said.. it errors out after I upload the file. 
If I delete just that, it works fine. 
Really confusing. lol

**Reply by bassamanator** [reply]
Show me the out of the the following:
```
cd ~/printer_data/config
tree
```

**Reply by bassamanator** [reply]
`sudo apt install tree` if need be.

**Reply by remote110** [reply]
.
├── cfgs
│   ├── adxl-direct.cfg
│   ├── adxl-rp2040.cfg
│   ├── adxl-rpi-pico-2x.cfg
│   ├── MECHANICAL_GANTRY_CALIBRATION.cfg
│   ├── misc-macros.cfg
│   ├── PARKING.cfg
│   └── TEST_SPEED.cfg
├── crowsnest.conf
├── KlipperScreen.conf
├── mainsail.cfg -> /home/biqu/mainsail-config/client.cfg
├── moonraker.conf
├── moonraker.conf.backup
├── printer-20230609_214210.cfg
├── printer-20230609_233722.cfg
├── printer-20230609_235054.cfg
├── printer-20230610_153749.cfg
├── printer-20230610_160157.cfg
├── printer.cfg
└── timelapse.cfg -> /home/biqu/moonraker-timelapse/scripts/klipper_macro/timelapse.cfg

1 directory, 19 files

**Reply by bassamanator** [reply, marked as answer]
You missed something in the `Before you begin` section, 5 or 6 point regarding `username`. Make the adjustments mentioned and you should be fine. Your username is `biqu` btw.

**Reply by remote110** [reply]
Thanks. I figured it was something I overlooked. 
I changed it to biqu in the printer.cfg but overlooked the other places.

**Reply by bassamanator** [reply]
Great! Please star the repo if you found my work useful.
